llvm/test/CodeGen/AIE/aie2/ra/waw_reg_renaming_loop.mir

```diff
@@ -0,0 +1,579 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py UTC_ARGS: --version 4
+# NOTE: Example file for Write After Write Register Renaming in Loop test
+#
+# This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# (c) Copyright 2024 Advanced Micro Devices, Inc. or its affiliates
+
+# RUN: llc -mtriple=aie2 -verify-machineinstrs --run-pass=greedy,aie-waw-reg-rewrite,virtregrewriter %s -o - | FileCheck %s
```

Comment:
I think it is better to rely on the pass pipeline itself here. For the first case:

`--start-before=greedy --stop-after=virtregrewriter`

Second case:

`--start-before=greedy --stop-after=virtregrewriter --aie-wawreg-rewrite=false`

In this way you make sure that your pass is correctly inserted within the pipeline.

Reply:
fixed